### PR TITLE
Fill form slots from form trigger message intent

### PR DESCRIPTION
**Pull request by smn-snkl**
@Ghostvv and myself discussed a way to allow form slots to be filled from the intent of the trigger message [here](https://forum.rasa.com/t/fill-form-slot-with-intent-of-initial-message/4744)

**Proposed changes**:
- Form event is now initialized with its trigger message
- Added a `trigger_message` attribute to `tracker.active_form`

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

**Comment by smn-snkl** [comment]
Respective changes in `rasa_core_sdk` can be found [here](https://github.com/RasaHQ/rasa_core_sdk/pull/44)

**Comment by smn-snkl** [comment]
@Ghostvv : Any idea why the `trigger_message` is stored in the tracker when running core in regular mode, but not in `interactive` mode?

**Comment by Ghostvv** [comment]
@smn-snkl thanks a lot for this work. I was on holiday, I'll review it this week

File: rasa_core/trackers.py [hunk 1]
```diff
@@ -145,12 +145,13 @@ def past_states(self, domain: 'Domain') -> deque:
         generated_states = domain.states_for_tracker_history(self)
         return deque((frozenset(s.items()) for s in generated_states))
 
-    def change_form_to(self, form_name: Text) -> None:
+    def change_form_to(self, form_name: Text, trigger_message: Text) -> None:
```
Comment (Ghostvv) [inline rasa_core/trackers.py]: I don't think `trigger_message` is a `Text` here

Comment (Ghostvv) [inline rasa_core/trackers.py]: Actually I think adding trigger message to the `Form` event is unnecessary. At this point `'trigger_message': self.latest_message`, no?

File: rasa_core/events/__init__.py [hunk 2]
```diff
@@ -835,8 +835,9 @@ class Form(Event):
     """
     type_name = "form"
 
-    def __init__(self, name, timestamp=None):
+    def __init__(self, name, trigger_message=None, timestamp=None):
```
Comment (Ghostvv) [inline rasa_core/events/__init__.py]: `trigger_message` is unnecessary here. See other comment

**Comment by smn-snkl** [comment]
@Ghostvv : Yea, I think so. That's what I initially meant on the [forum](https://forum.rasa.com/t/fill-form-slot-with-intent-of-initial-message/4744/36?u=smn-snkl). 

There we had
```
def change_form_to(self, form_name: Text) -> None:
    """Activate or deactivate a form"""
    if form_name is not None:
        self.active_form = {'name': form_name,
                            'validate': True,
                            'rejected': False,
                            'trigger_message': self.latest_message}
    else:
        self.active_form = {}
```
I think `self.latest_message` is enough though, right? Then you have access to all the relevant data

**Comment by Ghostvv** [comment]
sorry, I didn't get it from our discussion. Yes, but better put `self.latest_message.parse_data` directly so that is serializable. Like it is done here: https://github.com/RasaHQ/rasa_core/blob/7451990c9d3948709e4dc59f05f18d8be77e1a45/rasa_core/trackers.py#L132

**Comment by smn-snkl** [comment]
Ah, yes you're right. Sorry, I got `rasa_core` mixed up with `rasa_core_sdk` where `tracker.latest_message` is already the parsed one.

```
# latest_message is `parse_data`,
# which is a dict: {"intent": UserUttered.intent,
#                   "entities": UserUttered.entities,
#                   "text": text}
self.latest_message = latest_message if latest_message else {}
```

**Comment by Ghostvv** [comment]
@smn-snkl Hi, do you plan to continue working on this PR?

**Comment by smn-snkl** [comment]
@Ghostvv, yes generally I am. I’m a bit busy for a few more days. Would look into it then unless you say it’s very urgent?

**Comment by Ghostvv** [comment]
@smn-snkl thanks, for the update. It doesn't seem to be urgent right now.

**Comment by tmbo** [comment]
@smn-snkl we'll need to get this merged till next Tuesday as we will restructure the code afterwards and the code will go stale. I think the best thing would be to merge it before then. But if you don't have enough time to complete it, we'll need to close this PR and redoing the change will probably cost more time.

**Pull request closed without merging**